webapp/loggedin/modules/data/validationReport/validationReportView.js

```diff
@@ -0,0 +1,59 @@
+import './validationReportView.scss'
+
+import React from 'react'
+import { connect } from 'react-redux'
+
+import * as R from 'ramda'
+
+import * as Record from '@core/record/record'
+import * as NodeKeys from '@core/record/nodeKeys'
+
+import TableView from '@webapp/loggedin/tableViews/tableView'
+import { appModuleUri, dataModules } from '@webapp/loggedin/appModules'
+
+import { useOnUpdate } from '@webapp/commonComponents/hooks'
+
+import * as SurveyState from '@webapp/survey/surveyState'
+
+import { reloadListItems } from '@webapp/loggedin/tableViews/actions'
+
+import ValidationReportRowHeader from './validationReportRowHeader'
+import ValidationReportRow from './validationReportRow'
+
+const validationReportModule = 'validationReport'
+
+const ValidationReportView = ({ surveyCycleKey, reloadListItems, history }) => {
+  useOnUpdate(() => {
+    reloadListItems(validationReportModule, { cycle: surveyCycleKey })
+  }, [surveyCycleKey])
+
+  const onRowClick = record => {
+    const parentEntityUuid = R.prop(NodeKeys.keys.nodeUuid, R.last(record.keysHierarchy))
```

Comment:
I would create a method in NodeKeys to get keysHierarchy. it's not a good practice to access properties directly

Reply:
refactored almost entirely